src/darwin/Framework/CHIP/MTRUnfairLock.h

```diff
@@ -0,0 +1,46 @@
+/**
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * RAII wrapper around os_unfair_lock.
+ */
+
+#import <os/lock.h>
+
+class MTRAutoUnfairLock;
+
+class MTRUnfairLock
```

Comment:
Do we really need to wrap os_unfair_lock? MTRAutoUnfairLock can directly work with os_unfair_lock?

Reply:
We could do that; we just have to remember to init it.  But I can do that.

Reply:
OS_UNFAIR_LOCK_INIT is just 0, and ivars are default initialized to zero, so at least in ObjC classes there's actually nothing to do.

Reply:
Removed this wrapper.